### DISCUSSION TITLE
Order of Halves to Thirds

### DISCUSSION THREAD

**Discussion by notnotyz** [rxhanson/Rectangle#343] [Ideas]
The current order of Halves to Thirds is 1/2 -> 2/3 -> 1/3. Is it possible to change it to 1/2 -> 1/3 -> 2/3, which seems more intuitive? Many thanks!

**Comment by Grunerd** [comment]
This is the first thing I stumbled upon, coming from Spectacle. :D

Maybe we can make it optional, whether to start with the smaller or the larger window size first.

**Reply by rxhanson** [reply]
This is one of the things that I initially didn't understand about Spectacle - why the order was 1/2 -> 2/3 -> 1/3. Are you saying that for you the order in Spectacle was 1/2 -> 1/3 -> 2/3?

**Reply by notnotyz** [reply]
Haven't used Spectacle. But the order is 1/2 -> 1/3 -> 2/3 in hammerspoon MiroWindowsManager https://github.com/miromannino/miro-windows-manager.

**Comment by rxhanson** [comment]
I don't have the bandwidth to make this change at the moment, but I have no reasons to reject a pull request with this change, configurable via Terminal command.

**Reply by notnotyz** [reply]
Took a quick look at the source code. It seems simply swapping the 1 and 2 in lines 23 to 29 in [LeftRightHalfCalculation.swift](https://github.com/rxhanson/Rectangle/blob/90d6858dbfd26b581b9af6e12f10572f25618072/Rectangle/WindowCalculation/LeftRightHalfCalculation.swift) would work. i.e.
from
```
    func calculateSecondRect(_ params: RectCalculationParameters) -> RectResult {
        return calculateFractionalRect(params, fraction: 2 / 3.0)
    }
    
    func calculateThirdRect(_ params: RectCalculationParameters) -> RectResult {
        return calculateFractionalRect(params, fraction: 1 / 3.0)
    }
```

to
```
    func calculateSecondRect(_ params: RectCalculationParameters) -> RectResult {
        return calculateFractionalRect(params, fraction: 1 / 3.0)
    }
    
    func calculateThirdRect(_ params: RectCalculationParameters) -> RectResult {
        return calculateFractionalRect(params, fraction: 2 / 3.0)
    }
```

Does it make sense? Is it gonna break anything?

**Reply by rxhanson** [reply]
Yes, that's it. Not a big change, but to roll it into Rectangle, it'd have to be toggle-able using a Default. See the `Defaults` class; my preference here is to use an `OptionalBoolDefault`, so it would be called something like `Defaults.swappedRepeatedThirds.userEnabled`, and the defaults command would be setting it to int 1. I was going to add this into the previous release but forgot about it.

**Reply by notnotyz** [reply]
That is beyond something I can do in a short time. :) Greatly appreciate it! Once again, very neat app! I switched from MiroWindowsManager to this app. Big thanks!
